### PR TITLE
feat: v22 to v23 upgrade document

### LINKED ISSUE #1051: Migration guide V22 - V23
Create a migration guide informing the user what they may need to consder when upgrading from V22:
- [x] Use Java 11 or newer in the project (describe how to set the version in `pom.xml`). If using the `maven-failsafe-plugin` in your project, `jaxb` and other potential dependencies (https://maven.apache.org/surefire/maven-failsafe-plugin/java9.html) needs to be added (give example what to add to `pom.xml`).
- [x] The default package manager is now again `npm`. If you want to continue using `pnpm`, set `vaadin.pnpm.enable=true`.
- [x] If going with npm, ensure minimum npm version is 8.3 due to use of `overrides` section.
- [x] Node is now always used from `~/.vaadin` if global installation is incompatible

### PR DESCRIPTION
**Pull request by caalador**
Closes #1051

### CODE REVIEW

**Comment by tarekoraby** [comment]
To see the changes in the upgrade tool, I believe this PR still needs a couple of lines added to [upgrading/index.adoc](https://github.com/vaadin/docs/edit/latest/articles/upgrading/index.adoc).

Specifically, towards the end of the document, at line 192, the following should be added:
```
[.flow]
--
include::flow/_22-23.adoc[]
--
```

File: articles/upgrading/flow/_22-23.adoc [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    </dependencies>
+</plugin>
+----
+
+- Frontend package manager changes from pnpm to npm
++
+The package manager changes back to npm, but if using a version older than
+8.3 it is recommended to use pnpm instead.
++
+To keep using pnpm the configuration pnpm.enable should be set.
```
Comment (joheriks) [inline articles/upgrading/flow/_22-23.adoc:48]: Could mention here that the initial run of `npm` without a `package-lock.json` is slow. See https://github.com/vaadin/flow/issues/12825

**Review by joheriks — CHANGES_REQUESTED** [review]

File: articles/upgrading/flow/_22-23.adoc [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+vaadin {
+  pnpmEnable = true
+}
+----
++
+Or using the system property `vaadin.pnpm.enable=true`
++
+[NOTE]
+Running npm without a package-lock.json will be slow for the initial installation.
+Coming from a project on pnpm will not have a npm package-lock.json as it uses a package-lock.yaml
```
Comment (joheriks) [inline articles/upgrading/flow/_22-23.adoc:74]:
```suggestion
Coming from a project on pnpm will not have an npm `package-lock.json` as it uses `package-lock.yaml`
```

File: articles/upgrading/flow/_22-23.adoc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+vaadin {
+  pnpmEnable = true
+}
+----
++
+Or using the system property `vaadin.pnpm.enable=true`
++
+[NOTE]
+Running npm without a package-lock.json will be slow for the initial installation.
+Coming from a project on pnpm will not have an npm `package-lock.json` as it uses `package-lock.yaml`
```
Comment (joheriks) [inline articles/upgrading/flow/_22-23.adoc:74]:
```suggestion
Coming from a project on pnpm will not have an npm `package-lock.json` as it uses `pnpm-lock.yaml`
```

Sorry, one more.

**Review by joheriks — APPROVED** [review]

**Pull request merged**